src/main/java/com/checkmarx/flow/service/CodeBashingService.java

```diff
@@ -0,0 +1,130 @@
+package com.checkmarx.flow.service;
+
+import com.checkmarx.flow.config.CodebashingProperties;
+import com.checkmarx.flow.config.FlowProperties;
+import com.checkmarx.flow.constants.FlowConstants;
+import com.checkmarx.sdk.dto.ScanResults;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.omg.DynamicAny.DynAnyPackage.InvalidValue;
+import org.slf4j.Logger;
+import org.springframework.http.*;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+import javax.validation.ValidationException;
+import java.util.HashMap;
+import java.util.Map;
+
+@RequiredArgsConstructor
+@Service
+public class CodeBashingService {
+
+    private static Logger log = org.slf4j.LoggerFactory.getLogger(CodeBashingService.class);
+    private Map<String,String> lessonsMap = null;
+    private RestTemplate restTemplate = new RestTemplate();
+    private final FlowProperties flowProperties;
+    private final CodebashingProperties codebashingProperties;
+
+    public void createLessonsMap() {
+
+        try{
+            validateCodeBashingLessonsIntegration();
+            log.info("sending codebashing API [{}] request to get lessons map", codebashingProperties.getCodebashingApiUrl());
+            HttpEntity<?> httpEntity = new HttpEntity<>(createAuthHeaders());
+            ResponseEntity<String> response = restTemplate.exchange(codebashingProperties.getCodebashingApiUrl(), HttpMethod.GET, httpEntity, String.class);
+
+            if (response.getBody()==null){
+             log.error("can't get codebashing lessons. response is null");
+            }
+
+            log.info("codebashing API get lessons response: {} - {}", response.getStatusCode(), response.getStatusCodeValue());
+
+            JSONArray lessonsArray = new JSONArray(response.getBody());
+            lessonsMap = createLessonMapByCwe(lessonsArray);
```

Comment:
Here a state is introduced inside a service, which is generally not a good thing. This map should be returned to the outside and not stored inside this service instance.